File: python/longestCommonPrefix.py



class Solution:
    def commonPrefix(self, a: str, b: str) -> str:
        lengthA = len(a)
        lengthB = len(b)
        length = lengthA if lengthA >= lengthB else lengthB
        for i in range(length):
            if a[i] != b[i]:
                break
        return a[:i]

    def longestCommonPrefix(self, strs: List[str]) -> str:
        length = len(strs)
        if length == 0:
            return ""
        common = strs[0]
        for i in range(1, length):
            common = commonPrefix(strs[i], common)
            if common == "":
                return common
        return common
        
        

File: go/longestcommonPrefix.go
package leetcode

func longestCommonPrefix(strs []string) string {
	length := len(strs)
	if length == 0 {
		return ""
	}
	common := strs[0]
	for i := 1; i < length; i++ {
		common = commonPrefix(strs[i], common)
		if common == "" {
			return ""
		}
	}
	return common
}

func commonPrefix(a, b string) string {
	lengthA := len(a)
	lengthB := len(b)
	if lengthA == 0 || lengthB == 0 {
		return ""
	}
	length := 0
	if lengthA <= lengthB {
		length = lengthA
	} else {
		length = lengthB
	}
	i := 0
	for ; i < length; i++ {
		if a[i] != b[i] {
			break
		}
	}
	return a[0:i]
}
